### PR TITLE
Use Python.NET and Python.Included as externs

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

This will allow us to use custom builds of Python.NET for our CPython
engine. Projects now reference the binaries in 'extern' folder rather
than the published NuGets.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/Libraries/PythonNodeModels/PythonNodeModels.csproj [hunk 1]
```diff
@@ -39,13 +39,13 @@
       <HintPath>..\..\packages\Newtonsoft.Json.8.0.3\lib\net45\Newtonsoft.Json.dll</HintPath>
       <Private>False</Private>
     </Reference>
-    <Reference Include="Python.Included, Version=3.7.3.4, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Python.Included.3.7.3.4\lib\netstandard2.0\Python.Included.dll</HintPath>
-      <Private>False</Private>
```
Comment (mjkkirschner) [inline src/Libraries/PythonNodeModels/PythonNodeModels.csproj:44]: please retain the copyLocal / private elements set to False where they were previously or we'll get multiple copies of these dlls potentially pulled into various output folders.

**Comment by mmisol** [comment]
I had to retry the SelfServe job so I guess it got confused. The build passed and it is https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/805/

**Comment by mmisol** [comment]
https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/682/ also passed

**Comment by mmisol** [comment]
@mjkkirschner Can I ask you to take another look? The issues you pointed out should have been addressed. Thanks

File: test/Libraries/DynamoPythonTests/PythonEvalTests.cs [hunk 2]
```diff
@@ -165,7 +165,7 @@ public void CPythonEngineIncludesTraceBack()
             catch (Exception exc)
             {
                 // Trace back is not available for this call, but we still get a reasonable message back
-                Assert.AreEqual(@"SyntaxError : ('invalid syntax', ('<string>', 3, 13, ""print 'hello'\n""))", exc.Message);
+                Assert.AreEqual(@"SyntaxError : ('invalid syntax', ('<string>', 3, 7, ""print 'hello'\n""))", exc.Message);
```
Comment (mjkkirschner) [inline test/Libraries/DynamoPythonTests/PythonEvalTests.cs:168]: why did this change?

Comment (mmisol) [reply test/Libraries/DynamoPythonTests/PythonEvalTests.cs:168]: That's a good question. That's the type of error where the traceback is built-in. My guess is that was a result of the change in the version of Python, from 3.7 to 3.8, but I couldn't find it in the release notes. The column number changed to point to the beginning of the section where the invalid syntax was detected, so technically this seems more correct.

Comment (mmisol) [reply test/Libraries/DynamoPythonTests/PythonEvalTests.cs:168]: Maybe this https://bugs.python.org/issue38535?

**Pull request merged**